pkg/endpoints/credentials.go

```diff
@@ -152,6 +155,21 @@ func (r Resource) createCredential(request *restful.Request, response *restful.R
 		utils.RespondErrorAndMessage(response, err, errorMessage, http.StatusBadRequest)
 		return
 	}
+
+	// Patch service account
+	saName := ""
+	providedSaName := os.Getenv("PIPELINE_RUN_SERVICE_ACCOUNT")
```

Comment:
We're going to discuss this further next week but we're thinking this should be sent in the body for the request as opposed to getting it from here as it's something we potentially want to get from the user

Reply:
I see.  The create PipelineRun API will carry the service account name in the body.  So this api probably needs to do the same.

Reply:
The problem with requesting service account details from the user, is that we don't yet have an API to provide a list of service accounts. So this would be free text entry in the front end, which isn't ideal. I'd be ok with a hardcoded service account, at least until we have a better API around service accounts.

Reply:
This seems OK for now until we found issues for it.  Adding the service account name in the request causes that the service account name related to the secret must be persisted in somewhere.